pkg/osquery/tables/wlan/wlan.go

```diff
@@ -0,0 +1,198 @@
+package wlan
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const netshCmd = "netsh"
+
+var preludeInterfaceLineRegex = regexp.MustCompile(`Interface name :`)
+var preludeCountLineRegex = regexp.MustCompile(`There are [0-9]+ networks currently available`)
+
+type execer func(ctx context.Context, buf *bytes.Buffer) error
+
+type WlanTable struct {
+	client    *osquery.ExtensionManagerClient
+	logger    log.Logger
+	tableName string
+	getBytes  execer
+	parser    *OutputParser
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("name"),
+		table.TextColumn("authentication"),
+		table.IntegerColumn("signal_strength_percentage"),
+		table.TextColumn("bssid"),
+		table.TextColumn("radio_type"),
+		table.TextColumn("rssi"),
+		table.TextColumn("channel"),
+	}
+
+	parser := buildParser(logger)
+	t := &WlanTable{
+		client:    client,
+		logger:    logger,
+		tableName: "kolide_wifi_networks",
+		parser:    parser,
+		getBytes:  execCmd,
+	}
+
+	return table.NewPlugin(t.tableName, columns, t.generate)
+}
+
+func (t *WlanTable) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	var results []map[string]string
+	var output bytes.Buffer
+
+	err := t.getBytes(ctx, &output)
+	if err != nil {
+		return results, errors.Wrap(err, "getting raw wlan output")
+	}
+
+	scanner := bufio.NewScanner(&output)
+	scanner.Split(blankLineSplitter)
+	for scanner.Scan() {
+		chunk := scanner.Text()
+		if len(preludeInterfaceLineRegex.FindStringSubmatch(chunk)) > 0 {
+			continue
+		}
+		if len(preludeCountLineRegex.FindStringSubmatch(chunk)) > 0 {
+			continue
+		}
+
+		row := t.parser.Parse(bytes.NewBufferString(chunk))
+		if row != nil {
+			results = append(results, row)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		level.Debug(t.logger).Log("msg", "scanner error", "err", err)
+	}
+
+	return results, nil
+}
+
+func execCmd(ctx context.Context, buf *bytes.Buffer) error {
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	var stderr bytes.Buffer
+
+	args := []string{"wlan", "show", "networks", "mode=Bssid"}
+
+	cmd := exec.CommandContext(ctx, netshCmd, args...)
+	cmd.Stdout = buf
+	cmd.Stderr = &stderr
+
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "calling netsh wlan, got: %s", stderr.String())
+	}
+
+	return nil
+}
+
+func buildParser(logger log.Logger) *OutputParser {
+	return NewParser(logger,
+		[]Matcher{
+			{
+				Match: func(in string) bool {
```

Comment:
These will probably fail in other localizations.

Reply:
Yea this is a really good call-out. I _think_ that with the new approach (not depending on the output of `netsh wlan show networks`) this isn't an issue. The column names are defined in code.